content/rationale.md

```diff
@@ -0,0 +1,85 @@
+title: Why complate?
+
+Why complate?
+=============
+
+## Abstraction
+
+When we are developing our frontend components, we can use complate markup
+abstractions ("macros") as an abstraction layer for our component. Consider
+the following macro:
+
+```jsx
+function Navbar (_props, ...children) {
+  return <nav>
+    {children}
+  </nav>
+}
+```
+
+We can then use our macro in our application and it will be replaced with the
+markup we have defined.
+
+```jsx
+<Navbar />
+```
+
+This is _extremely powerful_ and solves a major pain point when writing
+maintainable frontend code. Requirements change constantly. When a new request
+comes, or the corporate design has changed yet again, I can change the markup
+of my component _without breaking any part of my application that uses the
+component_.
+
+Another major benefit is that it is possible to write complex frontend
+components and others can _use_ the components easily, without having to
+understand how the component works under the hood.
```

Comment:
I'm always careful to add that I'm not trying to shield users from the underlying markup; they should still be aware of that markup and know why it matters.

Reply:
Good point.

I just know that some people on my team love complate, because they can use existing components and make pretty UIs without actually having to write the CSS themselves.

Reply:
Yeah, I know, same here. But deep inside, I remain a naive, quixotic idealist...